### ISSUE TITLE
plot.rsk should have smarter `which` functionality

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#1022]

**Comment by dankelley** [comment]
Um, empty description. Were you going to assign yourself, Clark?  If so, empty is fine ... but if it's for me, I might need a tad more info 😆 (apparently that's the GH icon for laughing)

**Comment by richardsc** [comment]
Haha, yes, sorry. Assigned to me. I typed it out right in the middle of an oce tutorial with someone, and didn't have time to fill in any details.

Basically, the `which=` argument in `plot.rsk` only accepts a few options, dating back to when rsk objects only had pressure and temperature. I'm going to overhaul it so that it will accept any character that exists in the `@data` slot, much like `plotProfile()`

**Comment by richardsc** [comment]
Started looking through the code a bit today, and I think it's worth giving it a bit of an overhaul. Most of the options for `which=` are outdated as they were based on the original TDR files that we were working with.

I think the best scheme would be to do `which='timeseries'` (as is done now), and otherwise allow the `which` argument to correspond to the name of any of the fields in the `@data` slot. In each case the plot produced is still a time series. I don't think it makes sense to make profiles for `rsk` objects -- that's what `as.ctd()` is for.

**Comment by dankelley** [comment]
Right, your ideas make a lot of sense.

**Comment by richardsc** [comment]
I've started some work on this in my fork at https://github.com/richardsc/oce. Still a bunch to do and lots of testing, but a start.

**Comment by richardsc** [comment]
I've made some progress on this, but the one thing I'm a little unsure about is what to do with the `ylim` arg. For context, the "old" version of the rsk plot method required specifying `plim` and `Tlim`, but this was based on the (now false) premise that rsk objects only had pressure and temperature. I have removed those two arguments, but it makes no sense to try and code in a specific argument for *every* possible field that could exist in an rsk object.

So, for now, there is a `ylim` argument. However, that argument is ignored for the default plot, `which="timeseries"`, and only works if `plot.rsk()` is used to make a single panel plot, e.g.:
```r
library(oce)
d <- read.oce('create_data/rsk/060130_20150904_1159.rsk')
plot(d, which='temperature', ylim=c(2, 4))
```
If one wanted to make a multi-panel plot, there would have to be some way of specifying the `ylim` accordingly. E.g., if `length(which) > 1` then `ylim` can be specified as a matrix, something like:
```r
plot(d, which=c('temperature', 'pressure'), ylim=rbind(c(2, 4), c(20, 30)))
```
but that seems pretty confusing. An alternative way to do that with the `ylim` functionality I have already coded would be:
```r
par(mfrow=c(2, 1))
plot(d, which='temperature', ylim=c(2, 4))
plot(d, which='pressure', ylim=c(20, 30))
```
which is more typing, but arguably clearer.

Any thoughts @dankelley ?

**Comment by dankelley** [comment]
Of the two or three things I regret in `oce`, a big one is the ability to have `length(which)>1`. That has caused a lot of bugs and argument systems that are likely to confuse a user. In a normal Rgraph, `xlim` and `ylim` are pretty straightforward (as in your example).  Another problem that has come up with `length(which)>1` is that `adorn` issue. Really, it's just a mess. I think users will be happy to make 3 calls for 3 panels, because it lets them do whatever they want after each panel (heck, they might want to use `locator` on a panel, and the results of that might be used in the next panel, for a zoom or something...).

Since `plot,rsk-method` is being refactored quite significantly, maybe we just need to establish a convention: `which` should contain just 1 element.

Oce is still at version 0.something, so I actually think it would make sense to start changing other things (but only on code we use everyday, so we'll find errors). This will not change any defaults, e.g. `plot(ctd)` with no `which` argument specified would just make 4 calls.

I somehow doubt that many users are using multiple-`which` calls, since I don't recall any bug reports that mentioned this.

Radical, I know. Thoughts, @richardsc?

PS. probably won't read any answer until tomorrow afternoon.

**Comment by richardsc** [comment]
I think that's a good discussion to have, and it obviously relates to the present Issue, but I'm going to break it out so I can finish and close this one. Based on your opinion, it seems like you're ok with the functionality I describe above -- that to get `ylim` control the user has to make a series of individual calls. I've modified `plot,rsk-method` to work like this now, updated the documentation, and included an example to show how to do it.

One other thing I added was to put units in the yaxis labels. I know that I'm not doing this consistently with how it is done in other places (I think using `resizableLabel()`), but I don't fully understand the implementation, and `plot.rsk()` is not meant for final plot but for quick summary plots. It actually took me quite awhile to figure out all the expression/paste/bquote syntax, and once I got it I felt like the smaRtest pRogRammeR around:
```r
if (!is.null(x[['units']])) {
    unit <- x[['units']][[name]]$unit
    tmp <- c(name, '~"["*', as.character(unit), '*"]"')
    label <- bquote(.(parse(text=paste0(tmp, collapse=''))))
} else {
    label <- name
}
oce.plot.ts(x[["time"]], x[[name]], ylab=label, ...)
```

I'll make a pull request from my fork so that @dankelley can review when he has the chance.

**Comment by richardsc** [comment]
See PR #1084

**Comment by dankelley** [comment]
Yes, you are a smaRt pRogRammeR (man that's hard to type)! This bquote/parse stuff is tricky stuff to remember; I have to look it up every time.

**Comment by richardsc** [comment]
I'm going to take your acknowledgment of my programming skills as confidence that you think my PR is ok, and will merge it :). Feel free to undo or fix anything that looks wonky if you get a chance to review, but I've tested it a fair bit today and think it's better than the previous version of rsk plotting.

**Comment by richardsc** [comment]
Merged PR #1084. Closing

**Comment by dankelley** [comment]
Sorry, somehow I didn't see that it was a PR.  I need to start looking more closely at these things! I guess, more simply, I should just start looking to see if there is a number in the PR tab.  Thanks!
